anomalous-tx-value/src/agent.spec.ts

```diff
@@ -1,79 +1,64 @@
   FindingSeverity,
   Finding,
-  EventType,
-  Network,
-  HandleTransaction
+  HandleTransaction,
+  Network
 } from "forta-agent";
+import { NetworkManager } from "forta-agent-tools";
+import { TestTransactionEvent } from "forta-agent-tools/lib/test";
+import { provideHandleTransaction } from "./agent";
+import { AgentConfig, NetworkData } from "./utils";
 
-import agent, { DECIMALS, TX_VALUE_THRESHOLD } from ".";
+const DECIMALS = 10 ** 18;
+const DEFAULT_CONFIG: AgentConfig = {
+  [Network.MAINNET]: {
+    threshold: `${100 * DECIMALS}`
+  }
+};
 
 describe("Detect Very High Txn Value", () => {
   let handleTransaction: HandleTransaction;
+  let networkManager: NetworkManager<NetworkData>;
 
   beforeAll(() => {
-    handleTransaction = agent.handleTransaction;
+    networkManager = new NetworkManager(DEFAULT_CONFIG, Network.MAINNET);
+    handleTransaction = provideHandleTransaction(networkManager);
   });
 
-  const createTxEvent = ({
-    transaction,
-    addresses,
-    blockNumber
-  }: any): TransactionEvent => {
-    const tx: any = {
-      value: transaction.value
-    };
-    const receipt: any = {};
-    const block: any = { number: blockNumber };
-    return new TransactionEvent(
-      EventType.BLOCK,
-      Network.MAINNET,
-      tx,
-      receipt,
-      [],
-      addresses,
-      block
-    );
-  };
-
   describe("Handle Transaction", () => {
     it("returns empty findings if value is below threshold", async () => {
-      const txEvent = createTxEvent({
-        transaction: { value: 1 * DECIMALS }
-      });
+      const txEvent = new TestTransactionEvent();
 
+      txEvent.setValue(`${1 * DECIMALS}`);
       const findings = await handleTransaction(txEvent);
 
       expect(findings).toStrictEqual([]);
     });
 
     it("returns empty findings if value is equal to threshold", async () => {
-      const txEvent = createTxEvent({
-        transaction: { value: TX_VALUE_THRESHOLD }
-      });
+      const txEvent = new TestTransactionEvent();
 
-      const findings = await handleTransaction(txEvent);
+      txEvent.setValue(`${100 * DECIMALS}`);
 
+      const findings = await handleTransaction(txEvent);
       expect(findings).toStrictEqual([]);
     });
 
     it("returns a findings if value is above threshold", async () => {
+      const txEvent = new TestTransactionEvent();
       const value = 101 * DECIMALS;
-      const txEvent = createTxEvent({
-        transaction: { value: value }
-      });
 
-      const findings = await handleTransaction(txEvent);
+      txEvent.setValue(`${value}`);
 
+      const findings = await handleTransaction(txEvent);
       expect(findings).toStrictEqual([
         Finding.fromObject({
           name: "High Value Use Detection",
           description: "High value is used.",
           alertId: "NETHFORTA-2",
-          severity: FindingSeverity.High,
-          type: FindingType.Suspicious,
+          severity: FindingSeverity.Info,
+          type: FindingType.Info,
```

Comment:
Although it makes more sense to have them as "Info", I think we can leave them as they were in case the alerts are already used by other bots, perhaps filtered by severity/type

Reply:
makes sense 👍